BankLibrary/Bank.cs

```diff
@@ -9,18 +10,74 @@ public class Bank<T> where T : Account
 
         public void OpenAccount(OpenAccountParameters parameters)
         {
-            // TODO: check types compatibility
-            CreateAccount(parameters.AccountCreated, () => parameters.Type == AccountType.Deposit 
+            CheckValidTypeAccount(parameters.Type);
+
+            CreateAccount(parameters, () => parameters.Type == AccountType.Deposit
                 ? new DepositAccount(parameters.Amount) as T
                 : new OnDemandAccount(parameters.Amount) as T);
         }
 
-        private void CreateAccount(AccountCreated accountCreated, Func<T> creator)
+        public void Withdraw(decimal amount, int id)
+        {
+            var account = GetAccountById(id);
+
+            if (account is DepositAccount depositAccount)
```

Comment:
method `Withdraw` is defined in the base class, it should work without this if-else condition